starboard/build/config/BUILDCONFIG.gn

```diff
@@ -354,12 +369,65 @@ template("executable") {
 }
 
 template("shared_library") {
```

Comment:
It's mostly about cleaning this up, esp having certain invokers use `build_loader = false` to disable it